src/hardware/siglent-sds/protocol.c

```diff
@@ -519,10 +617,12 @@ SR_PRIV int siglent_sds_receive(int fd, int revents, void *cb_data)
 			switch (devc->model->series->protocol) {
 			case NON_SPO_MODEL:
 			case SPO_MODEL:
-				/* The older models need more time to prepare the the output buffers due to CPU speed. */
-				wait = (devc->memory_depth_analog * 2.5);
-				sr_dbg("Waiting %.f0 ms for device to prepare the output buffers", wait / 1000);
-				g_usleep(wait);
+				if (strcmp(devc->model->series->name, "SDS2000X+")) {
+					/* The older models need more time to prepare the the output buffers due to CPU speed. */
+					wait = (devc->memory_depth_analog * 2.5);
+					sr_dbg("Waiting %.f ms for device to prepare the output buffers", wait / 1000);
+					g_usleep(wait);
```

Comment:
I always dislike  sleep/wait calls. Is there no way to tweak SCPI read timeouts ?